_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,234 @@ also change the name of the columns in the DataFrame like this:
+> previous example, adding  `plt.show()` after the creation of the plot will enable your 
+> colleagues who aren't using a Jupyter notebook to reproduce your work on their platform.
+{: .callout}
 
-    plt.show()
+~~~
+fig, ax = plt.subplots()  # initiate an empty figure and axis matplotlib object
+ax.plot(x, y, '-')
 ~~~
 {: .language-python}
 
-We can include a legend by adding the optional keyword argument `label=''` in
-`plot()`. Caution: We cannot add labels to multiple lines that are plotted
-simultaneously by the `plt.plot()` command like we did above because Python
-won't know to which line to assign the value of the argument label. Multiple
-lines can also be plotted in the same figure by calling the `plot()` function
-several times:
+![line plot](../fig/08_line_plot.png)
 
-~~~
-    # Red dashes with no symbols, blue squares with a solid line, and green triangles with a dotted line
-    plt.plot(t, t, 'r--', label='linear')
-    plt.plot(t, t**2, 'bs-', label='square')
-    plt.plot(t, t**3, 'g^:', label='cubic')
+Although the latter approach requires a little bit more code to create the same plot, the advantage is that we now have **full control** of where the plot axis are placed, and we can easily add new items or, for example, more than one axis to the figure and adapt the labels:
 
-    plt.legend(loc='upper left', shadow=True, fontsize='x-large')
+~~~
+fig, ax1 = plt.subplots() # prepare a matplotlib figure
+ax1.plot(x, y, '-')
 
-    plt.xlabel('This is the x axis')
-    plt.ylabel('This is the y axis')
-    plt.title('This is the figure title')
+# adapt the labels
+ax1.set_ylabel('y')
+ax1.set_xlabel('x')
 
-    plt.show()
+# add an additional axis to the figure
+ax2 = fig.add_axes([0.2, 0.5, 0.4, 0.3]) # inset axes
+ax2.plot(x, y*2, 'r-')
 ~~~
 {: .language-python}
 
-The function `legend()` adds a legend to the figure, and the optional keyword
-arguments change its style. By default [typing just `plt.legend()`], the legend
-is on the upper right corner and has no shadow.
+![Additional axis to line plot](../fig/08_line_plot_inset.png)
 
- The functions `xlabel`, `ylabel`, `title`, `legend`, and many others create text labels. It is good to know that, in addition to the plain text, you may use mathematical notation using a subset of LaTeX language. See [this link](https://matplotlib.org/users/mathtext.html) for more information.
+### Link matplotlib, Pandas and plotnine
 
-
-Like MATLAB, pyplot is stateful; it keeps track of the current figure and
-plotting area, and any plotting functions are directed to those axes. To make
-more than one figure, we use the command `plt.figure()` with an increasing
-figure number inside the parentheses:
+When we create a plot using pandas or plotnine, both libraries use matplotlib 
+to create those plots. The plots created in pandas or plotnine are matplotlib 
+objects, which enables us to use some of the advanced plotting options available 
```

Comment:
I would keep them as they are as it explains the connection a bit more than just stating *the objects are matplotlib*. Maybe the second sentence can be removed, if this would improve the flow of the text. As you prefer, feel free to remove